.github/workflows/chromatic.yml

```diff
@@ -0,0 +1,42 @@
+# .github/workflows/chromatic.yml
+
+# Workflow name
+name: Chromatic Publish and Testing
+
+# Event for the workflow
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - ready_for_review
+    paths:
+      # Only run on file changes in any of these paths
+      - "src/components/**/*"
+      - "src/pages/**/*"
+      - "src/pages-conditional/**/*"
+      - "src/templates/**/*"
+
+# List of jobs
+jobs:
+  chromatic-deployment:
+    # Operating System
+    runs-on: ubuntu-latest
+    # Job steps
+    steps:
+      # 👇 Version 2 of the action
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # 👈 Required to retrieve git history
```

Comment:
See Chromatic docs on [Support for actions/checkout@v2 and above](https://www.chromatic.com/docs/github-actions#support-for-codeactionscheckoutv2code-and-above)